environment/src/main/java/com/sequenceiq/environment/environment/service/EnvironmentCreationService.java

```diff
@@ -106,6 +106,8 @@ public EnvironmentDto create(EnvironmentCreationDto creationDto) {
             // To keep backward compatibility, if somebody passes the group name, then we shall just use it
             environmentService.setAdminGroupName(environment, creationDto.getAdminGroupName());
         }
+        virtualGroupService.assignEnvironmentAdminRole(creationDto.getCreator(), environmentCrnForVirtualGroups);
```

Comment:
This is not related to virtual group at all. It should be moved to some other proper place.

Reply:
I agree, assignment logic should be in the service of the resource, in this case in some EnvironmentService class I assume

Reply:
Moved to the environmentService